homeassistant/components/deluge/sensor.py

```diff
@@ -1,128 +1,100 @@
-    from deluge_client import DelugeRPCClient
-
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    port = config.get(CONF_PORT)
-
-    deluge_api = DelugeRPCClient(host, port, username, password)
-    try:
-        deluge_api.connect()
-    except ConnectionRefusedError:
-        _LOGGER.error("Connection to Deluge Daemon failed")
-        raise PlatformNotReady
+
+
+async def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Import config from configuration.yaml."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Deluge sensors."""
+
+    client = hass.data[DOMAIN][config_entry.entry_id]
+    name = config_entry.data[CONF_NAME]
+
     dev = []
-    for variable in config[CONF_MONITORED_VARIABLES]:
-        dev.append(DelugeSensor(variable, deluge_api, name))
+    for sensor_type in SENSOR_TYPES:
+        dev.append(DelugeSensor(sensor_type, client, name,))
 
-    add_entities(dev)
+    async_add_entities(dev, True)
 
 
 class DelugeSensor(Entity):
     """Representation of a Deluge sensor."""
 
-    def __init__(self, sensor_type, deluge_client, client_name):
+    def __init__(self, sensor_type, client, client_name):
         """Initialize the sensor."""
         self._name = SENSOR_TYPES[sensor_type][0]
-        self.client = deluge_client
+        self.client = client
         self.type = sensor_type
         self.client_name = client_name
         self._state = None
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
         self.data = None
-        self._available = False
+        self.unsub_dispatcher = None
 
     @property
     def name(self):
         """Return the name of the sensor."""
         return f"{self.client_name} {self._name}"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the entity."""
+        return f"{self.client.api.host}-{self.name}"
+
     @property
     def state(self):
         """Return the state of the sensor."""
         return self._state
 
+    @property
+    def should_poll(self):
+        """Return the polling requirement for this sensor."""
+        return False
+
     @property
     def available(self):
         """Return true if device is available."""
-        return self._available
+        return self.client.api.available
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        self.unsub_dispatcher = async_dispatcher_connect(
+            self.hass, self.client.api.signal_update, self._schedule_immediate_update,
+        )
+
+    @callback
+    def _schedule_immediate_update(self):
+        self.async_schedule_update_ha_state(True)
+
     def update(self):
```

Comment:
Change this to `async async_update(self):` because no I/O is being done anymore, right ?

Reply:
yes the `update` method is called after I/O is done. not sure how you want to change the update method to `async_update`

Reply:
Never mind, done in last commit.